File: app/src/main/java/lt/setkus/interviewtest/configuration/dagger/component/ActivityComponent.java
package lt.setkus.interviewtest.configuration.dagger.component;

import android.app.Activity;

import dagger.Component;
import lt.setkus.interviewtest.configuration.dagger.PerActivity;
import lt.setkus.interviewtest.configuration.dagger.module.ActivityModule;

/**
 * @author <a href="mailto:robertas.setkus@gmail.com">robertas</a>
 */
@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {
    Activity activity();
}

File: app/src/main/kotlin/eu/oora/marvel/dependency/component/ActivityComponent.kt
package eu.oora.marvel.dependency.component

import dagger.Component
import eu.oora.marvel.MainActivity
import eu.oora.marvel.dependency.module.ActivityModule
import eu.oora.marvel.dependency.scope.PerActivityScope

@PerActivityScope
@Component(
  dependencies = arrayOf(
    ApplicationComponent::class
  ),
  modules = arrayOf(
    ActivityModule::class
  )
)
interface ActivityComponent : ApplicationComponent {
  fun inject(activity: MainActivity)
}
